x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryFolder.java

```diff
@@ -600,10 +600,12 @@ else if (target.foldable()) {
                 else {
                     GroupByKey matchingGroup = null;
                     if (groupingContext != null) {
-                        matchingGroup = groupingContext.groupFor(target);
+                        final Expression resolvedTarget = queryC.aliases().getOrDefault(target, target);
```

Comment:
This bit of code is confusing due to the left overs.
Target doesn't seem to be used anywhere or at least its initial value is no good - instead of adding a `resolvedTarget` just reinitialize it:
`target = queryC.aliases().getOrDefault(target, target)`

Same for `id`.

Reply:
Done